src/Interpreters/Aggregator.h

```diff
@@ -283,6 +331,37 @@ struct AggregationMethodStringNoCache
     }
 };
 
+template <typename TData>
+struct AggregationMethodStringNullableNoCache
```

Comment:
could we reuse existing AggregationMethods code (as well as HashMethods) through a private inheritance for example?

Reply:
I added a nullable generic parameter to replace the new struct type